yogstation/code/modules/antagonists/darkspawn/darkspawn_abilities/core_abilities.dm

```diff
@@ -0,0 +1,364 @@
+//////////////////////////////////////////////////////////////////////////
+/datum/action/cooldown/spell/touch/silver_tongue
+	name = "Silver Tongue"
+	desc = "When used near a communications console, allows you to forcefully transmit a message to Central Command, initiating a shuttle recall. Only usable if the shuttle is inbound. Costs 60 Psi."
+	button_icon_state = "silver_tongue"
+	button_icon = 'yogstation/icons/mob/actions/actions_darkspawn.dmi'
+	background_icon_state = "bg_alien"
+	overlay_icon_state = "bg_alien_border"
+	buttontooltipstyle = "alien"
+	invocation_type = INVOCATION_NONE
+	antimagic_flags = NONE
+	panel = "Darkspawn"
+	check_flags = AB_CHECK_CONSCIOUS | AB_CHECK_IMMOBILE
+	spell_requirements = NONE
+	psi_cost = 60
+	var/in_use = FALSE
+	hand_path = /obj/item/melee/touch_attack/darkspawn
+	var/duration = 8 SECONDS
+
+/datum/action/cooldown/spell/touch/silver_tongue/can_cast_spell(feedback)
+	if(SSshuttle.emergency.mode != SHUTTLE_CALL || in_use)
+		return
+	return ..()
+
+/datum/action/cooldown/spell/touch/silver_tongue/is_valid_target(atom/cast_on)
+	return istype(cast_on, /obj/machinery/computer/communications)
+	
+/datum/action/cooldown/spell/touch/silver_tongue/cast_on_hand_hit(obj/item/melee/touch_attack/hand, obj/machinery/computer/communications/target, mob/living/carbon/caster)
+	if(in_use)
+		return
+	if(target.stat)
+		to_chat(owner, span_warning("[target] is depowered."))
+		return FALSE
+	to_chat(owner, span_velvet("[pick("Pda ykw'lpwe skwo h'kccaz ej.", "Pda aiank'cajyu eo kran.", "Oknnu, bkn swop'ejc ukqn pkza.", "Wke swo kxn'znaz xu hws psk.")]"))
+	owner.visible_message(span_warning("[owner] briefly touches [src]'s screen, and the keys begin to move by themselves!"), span_velvet("You begin transmitting a recall message to Central Command..."))
+	play_recall_sounds(target)
+	in_use = TRUE
+	if(!do_after(owner, duration, target))
+		in_use = FALSE
+		return
+	in_use = FALSE
+	if(!target)
+		return
+	if(target.stat)
+		to_chat(owner, span_warning("[target] has lost power."))
+		return
+	SSshuttle.emergency.cancel()
+	to_chat(owner, span_velvet("The ruse was a success. The shuttle is on its way back."))
+	return TRUE
+
+/datum/action/cooldown/spell/touch/silver_tongue/proc/play_recall_sounds(obj/machinery/C) //neato sound effects
+	set waitfor = FALSE
+	playsound(C, "terminal_type", 50, TRUE)
+	for(var/i in 1 to ((duration/10)-1)) //8 seconds (80) -> 7 loops -> 7 seconds of random typing and beeping
+		sleep(1 SECONDS)
+		if(!C || C.stat || !in_use)
+			return
+		playsound(C, "terminal_type", 50, TRUE)
+		if(prob(25))
+			playsound(C, 'sound/machines/terminal_alert.ogg', 50, FALSE)
+			do_sparks(5, TRUE, get_turf(C))
+	sleep(0.4 SECONDS)
+	if(!C || C.stat || !in_use)
+		return
+	playsound(C, 'sound/machines/terminal_prompt.ogg', 50, FALSE)
+	sleep(0.4 SECONDS)
+	if(!C || C.stat || !in_use)
+		return
+	playsound(C, 'sound/machines/terminal_prompt_confirm.ogg', 50, FALSE)
```

Comment:
Would be cool if you can have a timer callback instead for each use instead of using sleep, especially since you're just re-checking the same thing each time.